### DISCUSSION TITLE
New Windows 10 Install, Chrome 87 64bit, cypress can't find chrome.  symlink doesnt fix it.

### DISCUSSION THREAD

**Discussion by finalcut** [cypress-io/cypress#9402] [Questions and Help] [answered]
Okay I've run through a lot of the troubleshooting guides so hopefully I can provide enough info here that it will spark a memory in someone and I can get this figured out.

I restored my laptop running Windows 10 two nights ago and then reinstalled chrome 87.0.4280.66 

I pulled down my project from git and tried to get my cypress tests to run in chrome using `npx cypress run --browser chrome` and I was told that chrome was not found.

Looking in the documentation I see that cypress expects it to be in `C:/Program Files (x86)/Google/Chrome/Application/chrome.exe` but it isn't since its a 64 bit install..   

I noticed this test that exist for cypress so I figured it should find it:

```js
exports['windows browser detection detects new Chrome 64-bit app path 1'] = {
  "name": "chrome",
  "version": "4.4.4",
  "path": "C:/Program Files/Google/Chrome/Application/chrome.exe"
}
```
But a `npx cypress info` doesn't list chrome.  Just edge shows up in the list.  

The [docs](https://docs.cypress.io/guides/references/troubleshooting.html#Windows) suggest I add a [symlink](https://www.howtogeek.com/howto/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/) (both HARD and not) to the official location `C:\Program Files\Google\Chrome\Application\chrome.exe` so I did that.  Cypress fails but it does find the browser.  

Here is the output of the `npx cypress info` command at that point:

```sh
1. Chrome
  - Name: chrome
  - Channel: stable
  - Version: 87.0.4280.66
  - Executable: C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
  - Profile: C:\Users\bill\AppData\Roaming\Cypress\cy\production\browsers\chrome-stable

2. Edge
  - Name: edge
  - Channel: stable
  - Version: 87.0.664.47
  - Executable: C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe
  - Profile: C:\Users\bill\AppData\Roaming\Cypress\cy\production\browsers\edge-stable
```

When I run `npx cypress run --browser chrome` I get the following error.

```sh
Running:  all\resource-table.spec.js                                                     (1 of 49)
spawn UNKNOWN
Error: spawn UNKNOWN
    at ChildProcess.spawn (internal/child_process.js:394:11)
    at Object.spawn (child_process.js:553:9)
    at Object.launch (C:\Users\bill\AppData\Local\Cypress\Cache\4.12.1\Cypress\resources\app\packages\launcher\lib\browsers.js:116:19)
    at Object.<anonymous> (C:\Users\bill\AppData\Local\Cypress\Cache\4.12.1\Cypress\resources\app\packages\server\lib\browsers\chrome.js:439:62)
    at step (C:\Users\bill\AppData\Local\Cypress\Cache\4.12.1\Cypress\resources\app\packages\server\lib\browsers\chrome.js:33:23)
    at Object.next 
[cypress-chrome.log](https://github.com/cypress-io/cypress/files/5624807/cypress-chrome.log)
(C:\Users\bill\AppData\Local\Cypress\Cache\4.12.1\Cypress\resources\app\packages\server\lib\browsers\chrome.js:14:53)
    at fulfilled (C:\Users\bill\AppData\Local\Cypress\Cache\4.12.1\Cypress\resources\app\packages\server\lib\browsers\chrome.js:5:58)
```

I'm going to attach the debug output log - its 540 lines.  

So .. anyone have any ideas?  (by the way if I install chrome canary that works but chrome still doesnt with cypress)

**Comment by bahmutov** [comment, marked as answer]
Could you try a newer version of Cypress, 4.12.1 is pretty old already.

**Reply by finalcut** [reply]
I will though I'm not sure if that will induce breaking changes into my tests at the moment :)

I'm also attempting to remove my hard link and rebuilding my WMI Repository first.. if that doesn't fix it I'll do newer cypress and let you know how that goes. 

Thanks

**Reply by finalcut** [reply]
It looks like 5.1.0 has a bugfix for my issue (I think) so I'll upgrade to that as a first level of upgrade before moving further.

Thanks

**Reply by finalcut** [reply]
5.1.0 fixed my problem and it only had a small breaking change I had to address 

```js
Cypress.Cookies.defaults({
  whitelist: cookie
})
```
to:
```js
Cypress.Cookies.defaults({
  preserve: cookie
})
```
